src/Symfony/Component/Security/Http/Authenticator/X509Authenticator.php

```diff
@@ -46,13 +48,13 @@ protected function extractUsername(Request $request): string
             $username = $request->server->get($this->userKey);
         } elseif (
             $request->server->has($this->credentialsKey)
-            && preg_match('#emailAddress=([^,/@]++@[^,/]++)#', $request->server->get($this->credentialsKey), $matches)
+            && preg_match('#'.$this->credentialUserIdentifier.'=([^,/]++)#', $request->server->get($this->credentialsKey), $matches)
```

Comment:
`preg_quote($this->credentialUserIdentifier, '#')` just in case

Reply:
Correct. I changed the highlighted line. This should be fine now, but let me add a test where a `#` is present.